### PR TITLE
(feat): InPageMenu and InPageMenuItem components

### LINKED ISSUE #66: InPageMenu and InPageMenuItem components
## Summary
Create InPageMenu and InPageMenuItem components in the component library

## Implementation
See designs. Just need this for the starter kit profile pages for now.

### PR DESCRIPTION
**Pull request by kieckhafer**
Resolves #66 
Impact: **minor**  
Type: **feature**

## Component
This PR adds the InPageMenu and InPageMenuItem components, for use as sub-navigation menus inside of pages.

## Screenshots
inpagemenu_ _reaction_design_system

## Breaking changes
None

## Testing
- See that the InPage Menu houses InPageMenuItems correctly
- See that `isSelected` styles are applied to InPageMenuItem correctly
- See that href / onClick works correctly

### CODE REVIEW

**Comment by machikoyasuda** [comment]
🚀 Preview deployed

Built with commit 9a855f52fc7761f2ea691abadf6479ed6fe3545a

https://deploy-preview-324--stoic-hodgkin-c0179e.netlify.com

**Comment by kieckhafer** [comment]
@spencern @aldeed 

Having a slight issue with getting linting to pass. Aside from this the PR is ready to be reviewed, what should I do about this, it doesn't really make sense...

On my local machine, I get this error:
1__reaction-component-library__bash_

And on circleci I get this error for the same lines:
continuous_integration_and_deployment

**Comment by kieckhafer** [comment]
Regarding the issue I was having above with linting, I went ahead and fixed it so it passes CircleCI, but it is not passing locally. That's something we should probably look into or try to figure out at some point.

File: package/src/components/InPageMenuItem/v1/InPageMenuItem.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     * set up a components context or you want to override one of the components in a
+     * single spot, you can pass in the components prop directly.
+     */
+    components: PropTypes.shape({
+      /**
+       * Pass either the Reaction iconExpand component or your own component that
+       * accepts compatible props.
+       */
+      iconExpand: PropTypes.node.isRequired,
+      Link: PropTypes.node.isRequired
```
Comment (nnnnat) [inline package/src/components/InPageMenuItem/v1/InPageMenuItem.js]: missing proptype docs

File: package/src/components/InPageMenu/v1/InPageMenu.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     * single spot, you can pass in the components prop directly.
+     */
+    components: PropTypes.shape({
+      /**
+       * Pass either the Reaction iconExpand component or your own component that
+       * accepts compatible props.
+       */
+      InPageMenuItem: PropTypes.node.isRequired
+    }).isRequired,
+    menuItems: PropTypes.arrayOf(PropTypes.object)
```
Comment (nnnnat) [inline package/src/components/InPageMenu/v1/InPageMenu.js]: missing proptypes doc

File: package/src/components/InPageMenuItem/v1/InPageMenuItem.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+     * Function to pass to button onClick
+     */
+    onClick: PropTypes.func
+  };
+
+  static defaultProps = {
+    isSelected: false
+  };
+
+  handleClick = preventAccidentalDoubleClick(() => {
```
Comment (nnnnat) [inline package/src/components/InPageMenuItem/v1/InPageMenuItem.js]: I think this method needs to do something like `event.preventDefault` or maybe the `href` doesn't need to be required. 

In the "With on click" example if you change the `href` prop to something like `href="/#!/thing"` then click the menu item the alert will fire but once it's closed the page will redirect to "/thing".

**Pull request merged**